src/arithmetic/arithmetic_expression.c

```diff
@@ -715,7 +396,43 @@ void AR_EXP_Free(AR_ExpNode *root) {
 	rm_free(root);
 }
 
-/* Mathematical functions - numeric */
+bool AR_FuncExists(const char *func_name) {
+	char lower_func_name[32] = {0};
+	size_t lower_func_name_len = 32;
+	_toLower(func_name, &lower_func_name[0], &lower_func_name_len);
+	void *f = TrieMap_Find(__aeRegisteredFuncs, lower_func_name, lower_func_name_len);
+	return (f != TRIEMAP_NOTFOUND);
+}
+
+void AR_RegisterFuncs() {
+	struct RegFunc {
+		const char *func_name;
+		AR_Func func_ptr;
+	};
+
+	struct RegFunc functions[39] = {
```

Comment:
If you don't want to track the function count explicitly, you can use the same logic as in the `_buildOperatorsWhitelist` function to dynamically size and iterate over this array.

Reply:
In this case the elements aren't pointers, not sure how to mark end of list